### ISSUE TITLE
podman support

### ISSUE THREAD

**Issue by anibal-aguila** [traefik/traefik#5730]
Traefik 2.0 Now GA! and want to know if have some plan to support [podman](https://github.com/containers/libpod)

Thanks in advance,

**Comment by SuperSandro2000** [comment]
It probably support podman already but no podman service discovery.

**Comment by kittydoor** [comment]
Is there any interest in supporting podman for service discovery?

**Comment by phozzy** [comment]
I agree with @SuperSandro2000 it is quite easy to run traefik using podman, the question is more about service discovery.
And here comes two variants:
* traefik and a container that traefik will forward requests to run inside one pod. So they share namespaces... is there any opportunity for serice discovery?
* traefik and containers that traefik will forward requests to run in separate containers. What can be used here for service discovery? systemd? I mean probably each container will have appropriate systemd service.

**Comment by NoX1De** [comment]
@kittydoor @phozzy Podman being **fully** supported as a backend is very desirable (including service discovery) in that case I believe both variants mentioned by @phozzy would be desirable.  Supporting podman properly with it being daemonless and also support considering that podman can run in rootless mode would all be desirable.  FWIW I believe podman is going to eventually supersede Docker in desirability for containerized development work as podman really eases the transition for organizations to eventually migrate/import things to orchestration platforms (being that it supports the concept of pods whereas Docker does not).

**Comment by bkircher** [comment]
Another reason why this would be nice to have is that current Docker CE on EL7 is broken somewhat because Red Hat decided to go with nftables in EL8 but docker does not really play nicely with that change (yet).

**Comment by neingeist** [comment]
podman's API: https://www.projectatomic.io/blog/2018/05/podman-varlink/

**Comment by tomkukral** [comment]
Varlink api will be deprecated and docker compatible API is going to land soon. See https://podman.io/blogs/2020/01/17/podman-new-api.html

**Comment by XakV** [comment]
Willing to help test, just following the issue.

**Comment by DnR-iData** [comment]
:+1: To get Podman fully supported as a backend.

Thanks a lot

**Comment by llabusch93** [comment]
Once Traefik will have full service discovery with podman, I say good bye to Docker. Highly interested in it as well, so +1!

**Comment by dtomcej** [comment]
Hello All,

We have discussed this a few times as a team, and have come to the following conclusions:

- Podman is not designed to provide any HA features itself, and it was built to be a drop in replacement for docker in Kubernetes environments

For this reason, connecting to podman for service discovery would be limited to a single machine, and that is definitely less enticing as a new development project to implement podman as a provider

- Podman's current API is accessible via VarLink

Varlink is pretty rough to work with, and although powerful, definitely makes the implementation a lot more difficult, and has far less rewards. Podman is moving to standard API bindings, which will make this point moot.

However, once the bindings are in place, it _should_ act the same way as the docker daemon, so it should be possible to re-use the docker Traefik provider, and just point the endpoint to podman. Note that this would only run on a single host, and would not allow for docker compose or swarm functionality, as those are not included in podman.

- Using Podman for service discovery

As with the first point, podman was not developed to be an orchestration platform, it leaves service management, as with many things to the overlying platform. The choice will be to either: Re-use the docker provider once a compatable API is released, or to use kubernetes, in which case podman is _already_ fully supported by Traefik, since Traefik interacts with kubernetes, not the runtime.

---
TL;DR

At the current point in time, writing a dedicated provider for podman does not provide any features or functionality that users would want to see.

Due to the current complexity of implementation, it would be a large time investment for minimal development

Podman has already noted that they are going to create a docker-compatable API, which will allow the existing Traefik Docker Provider to connect without having to rewrite a new provider.

Podman is currently supported when used with Kubernetes, which was its original goal.

I hope this gives some insight as to where we stand on this.

**Comment by tomkukral** [comment]
> * Podman is not designed to provide any HA features itself, and it was built to be a drop in replacement for docker in Kubernetes environments
> 

You are probably mixing up cri-o and podman. cri-o is designed to replace docker as container runtime in Kubernetes but podman is replacing docker itself (for running containers directly without Kubernetes).

Thank you for taking podman into account. I'm waiting for new version of podman with apiv2 (compatible with docker) and going to try replacing it.

**Comment by jpic** [comment]
Glad to see that podman now has a docker compatible API on /var/run/podman/podman.sock on arch linux provided by the podman systemctl service (systemd activated on podman socket).

But for now it's still emiting an empty configuration for me:

```
podman run -p 80:80 -v /var/run/podman/podman.sock:/var/run/docker.sock:ro traefik --providers.docker --log.lev
el=DEBUG 
time="2020-06-09T00:07:43Z" level=info msg="Configuration loaded from flags."
time="2020-06-09T00:07:43Z" level=info msg="Traefik version 2.2.1 built on 2020-04-29T18:02:09Z"
time="2020-06-09T00:07:43Z" level=debug msg="Static configuration loaded {\"global\":{\"checkNewVersion\":true},\"serversTransport\":{\"maxIdleConnsPerHost\":200},\"entryPoints\":{\"http\":{\"address\":\":80\",\"transport\":{\"lifeCycle\":{\"graceTimeOut\":10000000000},\"respondingTimeouts\":{\"idleTimeout\":180000000000}},\"forwardedHeaders\":{},\"http\":{}}},\"providers\":{\"providersThrottleDuration\":2000000000,\"docker\":{\"watch\":true,\"endpoint\":\"unix:///var/run/docker.sock\",\"defaultRule\":\"Host(`{{ normalize .Name }}`)\",\"exposedByDefault\":true,\"swarmModeRefreshSeconds\":15000000000}},\"log\":{\"level\":\"DEBUG\",\"format\":\"common\"}}"
time="2020-06-09T00:07:43Z" level=info msg="\nStats collection is disabled.\nHelp us improve Traefik by turning this feature on :)\nMore details on: https://docs.traefik.io/contributing/data-collection/\n"
time="2020-06-09T00:07:43Z" level=debug msg="Start TCP Server" entryPointName=http
time="2020-06-09T00:07:43Z" level=info msg="Starting provider aggregator.ProviderAggregator {}"
time="2020-06-09T00:07:43Z" level=info msg="Starting provider *traefik.Provider {}"
time="2020-06-09T00:07:43Z" level=info msg="Starting provider *docker.Provider {\"watch\":true,\"endpoint\":\"unix:///var/run/docker.sock\",\"defaultRule\":\"Host(`{{ normalize .Name }}`)\",\"exposedByDefault\":true,\"swarmModeRefreshSeconds\":15000000000}"
time="2020-06-09T00:07:43Z" level=debug msg="Configuration received from provider internal: {\"http\":{\"services\":{\"noop\":{}}},\"tcp\":{},\"tls\":{}}" providerName=internal
time="2020-06-09T00:07:43Z" level=debug msg="No default certificate, generating one"
time="2020-06-09T00:07:43Z" level=debug msg="Provider connection established with docker 1.9.3 (API 1.40)" providerName=docker
time="2020-06-09T00:07:43Z" level=debug msg="Configuration received from provider docker: {\"http\":{},\"tcp\":{},\"udp\":{}}" providerName=docker
time="2020-06-09T00:07:43Z" level=error msg="Provider connection error Error response from daemon: runtime error: index out of range [0] with length 0, retrying in 339.389208ms" providerName=docker
time="2020-06-09T00:07:43Z" level=debug msg="No default certificate, generating one"
time="2020-06-09T00:07:43Z" level=debug msg="Provider connection established with docker 1.9.3 (API 1.40)" providerName=docker
time="2020-06-09T00:07:43Z" level=debug msg="Configuration received from provider docker: {\"http\":{},\"tcp\":{},\"udp\":{}}" providerName=docker
time="2020-06-09T00:07:43Z" level=info msg="Skipping same configuration" providerName=docker
time="2020-06-09T00:07:43Z" level=error msg="Provider connection error Error response from daemon: runtime error: index out of range [0] with length 0, retrying in 1.06852642s" providerName=docker
time="2020-06-09T00:07:44Z" level=debug msg="Provider connection established with docker 1.9.3 (API 1.40)" providerName=docker
time="2020-06-09T00:07:44Z" level=debug msg="Configuration received from provider docker: {\"http\":{},\"tcp\":{},\"udp\":{}}" providerName=docker
time="2020-06-09T00:07:44Z" level=info msg="Skipping same configuration" providerName=docker
time="2020-06-09T00:07:44Z" level=error msg="Provider connection error Error response from daemon: runtime error: index out of range [0] with length 0, retrying in 1.530191392s" providerName=docker
^Ctime="2020-06-09T00:07:46Z" level=info msg="I have to go..."
time="2020-06-09T00:07:46Z" level=info msg="Stopping server gracefully"
time="2020-06-09T00:07:46Z" level=debug msg="Waiting 10s seconds before killing connections." entryPointName=http
time="2020-06-09T00:07:46Z" level=error msg="close tcp [::]:80: use of closed network connection" entryPointName=http
time="2020-06-09T00:07:46Z" level=debug msg="Entry point http closed" entryPointName=http
time="2020-06-09T00:07:46Z" level=info msg="Server stopped"
time="2020-06-09T00:07:46Z" level=info msg="Shutting down"
time="2020-06-09T00:07:46Z" level=error msg="accept tcp [::]:80: use of closed network connection" entryPointName=http
time="2020-06-09T00:07:46Z" level=debug msg="FIXME: Got an status-code for which error does not match any expected type!!!: -1" module=api status_code=-1
time="2020-06-09T00:07:46Z" level=error msg="Failed to retrieve information of the docker client and server host: error during connect: Get \"http://%2Fvar%2Frun%2Fdocker.sock/v1.24/version\": context canceled" providerName=docker
time="2020-06-09T00:07:46Z" level=error msg="Cannot connect to docker server error during connect: Get \"http://%2Fvar%2Frun%2Fdocker.sock/v1.24/version\": context canceled" providerName=docker
```

**Comment by mcnugit** [comment]
@jpic  did you solve this?

**Comment by jdoss** [comment]
I hope Podman support can be considered for development in the near future since Podman 2.0 is now GA https://podman.io/blogs/2020/06/29/podman-v2-announce.html

**Comment by jpic** [comment]
@mcnugit I haven't ! But @SantoDE assigned to the issue so I'm definitely hoping for some podman support in the following months ... It's just so much more fun than Docker !

**Comment by XakV** [comment]
Hope this is being revisited. Thanks for considering.

**Comment by jpic** [comment]
Untested but in theory a script that reads from `podman events` and extracts the traefik annotations from containers to add/remove to from the traefik file configuration with the directory watcher.

In this way it seems pretty easy, but not sure if that will work for cluster use cases.

**Comment by FedericoAntoniazzi** [comment]
Hi, any news?
Are there any WIP? I could try to contribute

**Comment by ldez** [comment]
https://github.com/traefik/traefik/issues/5730#issuecomment-594652919

**Comment by yangm97** [comment]
@ldez that comment is very outdated by now. The varlink API has been deprecated and podman now provides a REST API over a unix socket, just like docker. It also implements some of the docker endpoints so maybe Traefik already supports podman to some extent. For instance, I've had success pointing the docker-client to the podman socket, it allowed me to list and create containers, etc. just like as if it was talking to a docker daemon.

The real deal with supporting the podman API though would be to expose pods created from podman to traefik, since you probably won't get meaningful pod information by using the docker compatibility endpoints.

**Comment by rsommer** [comment]
This seems blocked by containers/podman#5580. I was trying to use traefik with podman via APIv2. The containers are discovered but the required network informations for actual routing to them are missing from the API endpoint.

**Comment by TerraTech** [comment]
I think this may have been fixed by containers/podman#9109  (?)

**Comment by brettinternet** [comment]
In case it's unclear to anyone that has arrived here (as it was to me at first), Traefik + Podman is compatible but you must use a [file provider](https://doc.traefik.io/traefik/providers/file/).

Here's [my file provider](https://github.com/brettinternet/homelab/blob/master/roles/traefik/templates/traefik.yml.j2) in an ansible role. I use Traefik with a [rootless Podman configuration](https://github.com/brettinternet/homelab/blob/master/roles/compose/tasks/traefik.yml) with just a simple, local orchestration.

There are real advantages to the service discovery provided by the Docker provider, but I wanted to call out the workaround.

**Comment by PixelJonas** [comment]
to be honest. "service discovery via labels" is one of the main reasons I'm using traefik and the /only/ reason I'm still on docker and did not move to podman.

**Comment by ADIX7** [comment]
Podman v3 will have a docker compatible api (I guess v2 does also have but it is not complete yet). You will be able to use docker-compose but the containers will start in podman if you enable the socket service.

https://www.redhat.com/sysadmin/podman-docker-compose
Take a look at the 2nd example.

**Comment by andrewsav-bt** [comment]
Looks like redhat is building full fledged docker replacement. With OpenShift and Kubernetes having ditched docker already this is not quite surprising.

This issues was opened a year and 3 month ago, and it was not clear back then what direction podman and co would take. It seems that in today's landscape it's more probable that trafik supports it some day _if_ we see wider movement from docker to podman.

**Comment by AndrewSav** [comment]
Looks like redhat is building full fledged docker replacement. With OpenShift and Kubernetes having ditched docker already this is not quite surprising.

This issues was opened a year and 3 month ago, and it was not clear back then what direction podman and co would take. It seems that in today's landscape it's more probable that trafik supports it some day if we see wider movement from docker to podman.

**Comment by ADIX7** [comment]
> > https://www.redhat.com/sysadmin/podman-docker-compose
> 
> Looks like RedHat is building full fledged docker replacement. With OpenShift and Kubernetes having ditched Docker already this is not quite surprising.
> 
> This issues was opened a year and thee months ago, and it was not clear back then what direction Podman and co would take. It seems that in today's landscape it's more probable that Traeik supports it some day if we see wider movement from Docker to Podman. And of course if Podman implements API identical to docker, Traefik itself won't even need to do much for support (apart may be from some testing).

As I understand they want a 100% docker compatible API. Podman advertised as a drop-in docker replacement and the CLI is also fully "docker compatible". They mentioned multiple times you can do `alias docker=podman` and good to go. In v3 the API will be mostly finished I guess. So Traefik should work with it. The ability to use docker-compose is a bonus but if docker compatibility is not neccessary for someone I would rather use the kube yaml synthax (well, maybe not yet cause it has some missing feature and bugs but in the near feature).

**Comment by PixelJonas** [comment]
@ADIX7  thanks for that link. I'll test it this weekend. This could finally replace my docker-compose setup :D

**Comment by MartinReinke** [comment]
@PixelJonas any outcomes from your home testing exercise?

**Comment by jpic** [comment]
You can make a trivial script that reads from docker events stout and
maintains a directory of configurations based on that.

Le mar. 2 févr. 2021 à 09:53, Jonas Janz <notifications@github.com> a
écrit :

**Comment by ilyazzz** [comment]
My experience using latest podman with traefik:

First, you need to configure podman to expose the docker socket as described in the article linked above. On debian/ubuntu you have to manually write in a tmpfiles.d entry as they don't provide a `podman-docker` compatibility package. Then you can set up traefik normally as you would with docker. I've personally used podman-compose instead of docker-compose, but either should be fine.

- You have to run traefik as a root container, since it can't access the "docker" socket when running rootless.
- Root container service discovery works if your container exposes ports or if you specify the port manually in traefik. I couldn't get whoami to expose itself via the API, but it could be a configuration issue on my part.

    Example of a working auto-discovered container:
```
version: '3'
    
services:
  nginx:
    image: nginx
    environment:
      - NGINX_PORT=80
      - NGINX_HOST=nginx-nginx-1
    labels:
      - "traefik.http.services.nginx.loadbalancer.server.port=80"
```
- Rootless containers don't get discovered, as they don't create their own network interface. They should work fine if you map the ports and specify them manually in the file provider though.

**Comment by thmo** [comment]
Can confirm that on CentOS8, with Podman 3.0.1 (installed from [here](https://build.opensuse.org/project/show/devel:kubic:libcontainers:stable), as described on https://podman.io/getting-started/installation) and `docker.io/library/traefik:v2.4`, things work as expected, running the container as root and giving it access to the Docker socket (which is a symlink to the Podman socket).

The only point to keep in mind is to explicitly announce a service's port using a label like this: `traefik.http.services.<servicename>.loadbalancer.server.port=<port>`.

**Comment by qm78** [comment]
Can confirm on CentOS8 Stream, with Podman 3.0.0-dev rootless containers fail to get discovered, due to container's ip address can't be found:

`level=error msg="service \"whoami\" error: unable to find the IP address for the container \"/whoami\": the server is ignored" providerName=docker container=whoami-...`

Further confirm rootless containers with file provider just work fine, but is not as elegant as using container labels. Any chance to get rootless containers configured by container labels some day?

**Comment by ilyazzz** [comment]
@qm78 as far as I'm aware there's no way to communicate with rootless containers without mapping ports. As rootless containers do not create their own network interfaces (they don't have permission to), you can't talk to them by their ip address.

It would probably be possible to define all the necessary options via labels (though I'm not sure how), it would still be traefik talking to some defined port on the host, rather than directly talking to containers without the containers exposing anything (as is the case with rootfull containers or with docker).

**Comment by thmo** [comment]
With podman 3.0.1 on Fedora 33 it is possible to create a network as user, and use that for your rootless containers. They will get an IP there. Tried starting rootless containers for traefik and whoami, both connected to that network:
```console
$ podman network create foo
$ systemctl --user start podman.service
$ podman run --network=foo --name traefik -d --rm -p 8080:80 --volume /run/user/$UID/podman/podman.sock:/var/run/docker.sock docker.io/library/traefik:v2.4 --providers.docker --entrypoints.http.address=:80
17f61674...
$ podman inspect -f '{{.NetworkSettings.Networks.foo.IPAddress}}' traefik 
10.88.2.7
$ podman run --network=foo --name whoami -d --rm --label 'traefik.http.services.whoami.loadbalancer.server.port=80' docker.io/containous/whoami
$ podman inspect -f '{{.NetworkSettings.Networks.foo.IPAddress}}' whoami
10.88.2.8
```
traefik will print an error that it couldn't get an IP for the "/rootless-cni-infra" container that is automatically created, but it will _not_ complain about the other containers. So this looked quite promising.

However, communicating with the whoami container doesn't really work:
```console
$ curl http://127.0.0.1:8080
curl: (56) Recv failure: Connection reset by peer
```

**Comment by ilyazzz** [comment]
Correct me if I'm wrong, but it looks like the rootless networks are only used for communication between containers, they don't create a network interface and you can't connect to them from the host:
```
$ podman inspect -f '{{.NetworkSettings.Networks.foo.IPAddress}}' some-rootless-container
10.88.2.2
$ ping 10.88.2.2
PING 10.88.2.2 (10.88.2.2) 56(84) bytes of data.
```
Meanwhile, rootfull containers get a network interface `vethXXXXXX@ifX`, and you can connect to them:
```
$ sudo podman inspect -f '{{ .NetworkSettings.IPAddress }}' some-rootfull-container
10.88.0.2
$ ping 10.88.0.2
PING 10.88.0.2 (10.88.0.2) 56(84) bytes of data.
64 bytes from 10.88.0.2: icmp_seq=1 ttl=64 time=0.039 ms
```

If your intention is to have traefk within the same podman network as other containers, that would probably work. But as far as I'm aware there's no way to communicate between rootless networks without binding ports to the host, and I believe that sharing a network between all services isn't desirable.

**Comment by AndrewSav** [comment]
>  that sharing a network between all services isn't desirable.

In docker you can have as many networks as you want, is it not the case with podman?

**Comment by ilyazzz** [comment]
> In docker you can have as many networks as you want, is it not the case with podman?

It is the case with podman, but communication between different networks is only possible when running podman as root (as is the case with docker). The previous comment referred to running rootless containers, which isn't a thing with docker in the first place.

**Comment by thmo** [comment]
> Correct me if I'm wrong, but it looks like the rootless networks are only used for communication between containers within that network

Yes, sure, but the "connection" to the outside is done via the exposed port of one of those containers, namely that running traefik. IIUC, that connection is realized via "conmon". All the rootless containers connected to the same network ("foo") can communicate with each other. This is all very similar now to the situation with rootfull containers.

Note that in my experiment from above, the `curl` command resulted in "Connection reset", not "Connection refused", hence at least _something_ is in place. So it's either a bug, or some additional config is needed.

**Comment by Yajo** [comment]
Possibly rootless networking won't be possible to support until https://github.com/containers/podman/issues/9169 is fixed.

Since podman runs under the same permission and process umbrella as the user who runs it, it's like having a docker daemon for each user. user_1 will have his own containers, networks, etc. root will also have his own. Of course, only root can do some specific stuff (such as creating fake network interfaces or listening on ports below 1024), just like would happen in the normal Linux world.

If you want to run unprivileged processes, with rootless containers, those will have to listen to unprivileged ports, and traefik will have to run as root to bind on ports 80 and 443, and will have to proxy to local ports. That's how life has been always before docker.

> which isn't a thing with docker in the first place

Yep, I think that the fact we're worrying about using root is why podman is good: it makes you realize what you're doing.

Docker has always been root-or-nothing, and since there was no other option, nobody cared. That was one of its dangers. So, even if you have to run podman as root, you're not worse than before after all.

**Comment by yangm97** [comment]
> and traefik will have to run as root to bind on ports 80 and 443

Actually you just need to add the capability for binding on lower ports in your sysctl (`sysctl net.ipv4.ip_unprivileged_port_start=0`) or launch a rootless container using root but hand it back to your user (`sudo podman run --user ${id -u} --cap-add cap_net_bind_service ...`).

**Comment by ilyazzz** [comment]
Update: with latest podman built from git, I got auto-discovery working with rootless containers. Though for some reason It only works with docker-compose, not podman-compose.
You need the podman user systemd service running for this to work. You also need to set `ip_unprivileged_port_start` if you want traefik to be able to listen on 80/443.

traefik's docker-compose:
```
version: "3"

services:
  reverse-proxy:
    image: traefik:v2.4
    command: --api.insecure=true --providers.docker
    ports:
      - "80:80"
      - "8080:8080"
    volumes:
      - /run/user/1000/podman/podman.sock:/var/run/docker.sock
```

Then it can be started with `docker-compose -H unix:/run/user/1000/podman/podman.sock up -d`.

A service:
```
version: "3"

services:
  whoami:
    # A container that exposes an API to show its IP address
    image: traefik/whoami
    labels:
      - "traefik.http.routers.whoami.rule=Host(`whoami.docker.localhost`)"
      - "traefik.http.services.whoami.loadbalancer.server.port=80"
```

After which, `curl -H Host:whoami.docker.localhost http://127.0.0.1` properly returns everything:

```
IP: 127.0.0.1
IP: 10.88.4.3
Host: whoami.docker.localhost
```

As you can see by the loadbalancer port being specified manually, port discovery via the API still doesn't work. Another quirk I've found is that services don't appear to get removed when their container shuts down (which is not a thing with normal rootfull containers).

**Comment by athisun** [comment]
As much as I wanted to get service discovery working, after much research I've found out that allowing access to the Docker/Podman socket completely negates the security benefits of rootless containers, even when binding it as a read-only volume. The socket effectively grants full control over the host, hence why SELinux prevents rootless containers from accessing it. If you did want to use service discovery, it seems you'd need to either disable SELinux ([but please don't](https://stopdisablingselinux.com/)) or run Traefik from a rootfull container.

See this bug report: https://bugzilla.redhat.com/show_bug.cgi?id=1495053#c2 and the security notes under https://doc.traefik.io/traefik/providers/docker/#docker-api-access

**Comment by PixelJonas** [comment]
I'm currently running (successfully) traefik in a rootfull container. But I just recently stumbled upon this blog-post:

https://chriswiegman.com/2019/11/protecting-your-docker-socket-with-traefik-2/

This may be a way to run traefik non-rootfull and use the tcp exposed docker-socket

**Comment by rtribotte** [comment]
Hello,

We are seeing more and more people using Traefik with Podman, which makes us happy and we want to move on and change our position regarding the Podman support, we just haven't had the capacity yet to fully confirm that it works well.

We are currently reviewing https://github.com/traefik/traefik/pull/9190 which adds support for the `host.containers.internal` host lookup to discover container IP in network host mode, to the Docker provider.
To that occasion, we have tested Traefik and our Docker provider with Podman, and aside from the need for the rights to access the socket, it seems to be working well.
But those tests were not conducted thoroughly, and we cannot conclude that the Docker provider is working well in every situation with Podman.

Therefore, our current stance is to close this issue and consider that the Docker provider can be used with Podman. Notably, because the preliminary tests show that it plays well for at least basic use cases, and because we are most probably going to introduce the support for the `host.containers.internal` host specific to Podman.

Nonetheless, we do not exclude changing our mind if Podman doesn't fit with the Docker provider, and these are the possible reasons that we can foresee at the moment:
- The Docker provider needs to evolve, and due to added and incompatible specificities will not be usable anymore with Podman
- Podman related bugs or features which would add too much complexity to the Docker provider

**Comment by XakV** [comment]
Thank you! Will test and open a separate issue if there are any rough edges.

-- 
  Zach Villers
  ***@***.***
